lib/context.js

```diff
@@ -88,24 +89,51 @@ class ResponseContext {
     const next = cb(this._next).once();
     const req = this._req;
     const res = this._res;
-    const returnURL = params.returnTo || this._config.baseURL;
+
+    const returnTo = params.returnTo || req.query.returnTo; 
+
+    if (
+      returnTo && 
+      !returnTo.startsWith(this._config.baseURL) && url.parse(returnTo).host &&
+      !this._config.postLogoutRedirectUris.includes(returnTo)
+    ) {
+      const err = new Error(`returnTo (${returnTo}) URI was not registered in config's postLogoutRedirectUris.`);
+      return next(err);
+    }  
+    
+    let returnURL = returnTo || this._config.postLogoutRedirectUri;
+    
+    if (url.parse(returnURL).host === null) {
+      returnURL = urlJoin(this._config.baseURL, returnURL);
+    }
+    
+    if (!req.session || !req.openid) {
```

Comment:
We removed the `req.session` requirement in 0.6.0 so we should not be checking that here. Use `req.isAuthenticated()`